### PR TITLE
Finish photometry notebooks

### PR DESCRIPTION
**Pull request by mwcraig**
This pull requests finishes off the notebooks needed to perform photometry in 2.0. 

Still to do:

- [x] Ensure coverage on new code is 100%
- [ ] Add new notebooks to launcher
- [ ] Check for an intermittent error when starting notebooks from the launcher. Some times jupyter lab gets very confused and can't display widgets. Hoping this was just a one-time thing. 

**NOTE**: 

- do not review this yet; currently it is unfinished and contains all of the changes in #371 
- #371 should get merged before this.

### CODE REVIEW

File: stellarphot/settings/custom_widgets.py [hunk 1]
```diff
@@ -985,3 +989,71 @@ def save_wd(_=None):
...
         )
+
+
+class PhotometryRunner(ipw.VBox):
+    def __init__(
+        self, photometry_notebook_name="photometry_run.ipynb", *args, **kwargs
+    ):
+        super().__init__(*args, **kwargs)
+        self.photometry_notebook_name = photometry_notebook_name
+        self.fo = FitsOpener(
```
Comment (JuanCab) [inline stellarphot/settings/custom_widgets.py:1000]: `fo` upset the spell checker, so I renamed a few of these to `fitsopen` or something like that in the past.  I can see why you added it to the ignore list.

File: stellarphot/settings/tests/test_photometry_runner.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+# so that the user_data_dir method returns the temporary directory.
+@pytest.fixture(autouse=True)
+def fake_settings_dir(mocker, tmp_path):
+    mocker.patch.object(
+        settings_files.PlatformDirs, "user_data_dir", tmp_path / "stellarphot"
+    )
+
+
+class TestPhotometryRunner:
+
```
Comment (JuanCab) [inline stellarphot/settings/tests/test_photometry_runner.py:28]: Spurious blank line after class definition?  I don't think [PEP8 recommends this](https://peps.python.org/pep-0008/#blank-lines) (feel free to disprove me, I am only about 70% sure)... should probably remove.

File: stellarphot/settings/tests/test_photometry_runner.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        fits_data.writeto(fake_fits)
+
+        photometry_runner = PhotometryRunner()
+        # Select a fits file
+        photometry_runner.fo.file_chooser.reset(".", fake_fits.name)
+        photometry_runner.fo.file_chooser._apply_selection()
+        photometry_runner.fo.file_chooser.value = fake_fits
+
+        # Check that the message in the information box is as expected
+        assert "Photometry will be done" in photometry_runner.info_box.value
```
Comment (JuanCab) [inline stellarphot/settings/tests/test_photometry_runner.py:78]: Presumably if you listened to my earlier recommendation to replace "done" with "performed" then this line of the test will need updating.

**Review by JuanCab — APPROVED** [review]
Only checked commit 85b87d forward, since others covered with PR #371  . No showstoppers but check the feedback. I didn't actually RUN the notebooks yet since this is a draft.  I will if you want me to check everything.

File: pyproject.toml [hunk 4]
```diff
@@ -25,6 +25,7 @@ dependencies = [
     "ipywidgets",
     "jupyter-app-launcher",
     "matplotlib",
+    "papermill",
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply pyproject.toml:28]: yes -- papermill launches a notebook with parameters in it, adds a cell with parameter values and saves the notebook.

File: stellarphot/notebooks/run-launcher.ipynb [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "e5f2fc8f-b893-400e-8fb7-4dd8e871334f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ipywidgets as ipw\n",
+    "\n",
+    "from stellarphot.settings.custom_widgets import Kaboodle"
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/notebooks/run-launcher.ipynb:12]: Fixed this...

File: stellarphot/settings/custom_widgets.py (hunk 1 above)
Comment (mwcraig) [reply stellarphot/settings/custom_widgets.py]: Fixed

File: stellarphot/settings/tests/test_photometry_runner.py (hunk 2 above)
Comment (mwcraig) [reply stellarphot/settings/tests/test_photometry_runner.py:28]: Seems to pass ruff either way so I deleted the blank line

File: stellarphot/settings/tests/test_photometry_runner.py (hunk 3 above)
Comment (mwcraig) [reply stellarphot/settings/tests/test_photometry_runner.py]: You were correct, sir. I didn't follow your advice initially, but you were correct...

**Comment by mwcraig** [comment]
@JuanCab -- this is ready for a look. Sorry it got a little long. Once this is merged it will be possible to perform photometry via notebooks.

I'll work on a separate PR to add some more documentation and another to make the command line approach is fairly straightforward.

I'm not sure why coverage thinks we need to test our test files....

**Review by JuanCab — APPROVED** [review]
I only reviewed commit 41416b3 (WIP photometry run settings) forward... and that is enough (24 commits)!

Most of the comments have to deal with the fact that Codecov was really freaking out demanding tests of tests and missing code that was getting tested.  A few of the comments are more question/informational.  No showstoppers and the tests look good.

File: stellarphot/gui_tools/tests/test_profile_and_comps.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+import pytest
+from astropy.utils.data import get_pkg_data_filename
+
+from stellarphot.gui_tools import ComparisonAndSeeing
+
+
+class TestComparisonAndSeeing:
+    def copy_fits_to_temp(self, tmp_path):
+        # Copy a fits file to a temporary directory
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/gui_tools/tests/test_profile_and_comps.py:11]: This isn't a test, it just takes care of copying the FITS file needed for a couple of the other tests in the class

**Pull request merged**